pkgs/by-name/bm/bmake/setup-hook.sh

```diff
@@ -42,11 +41,9 @@ bmakeCheckPhase() {
         local flagsArray=(
             ${enableParallelChecking:+-j${NIX_BUILD_CORES}}
             SHELL="$SHELL"
-            # Old bash empty array hack
-            $makeFlags ${makeFlagsArray+"${makeFlagsArray[@]}"}
-            ${checkFlags:-VERBOSE=y} ${checkFlagsArray+"${checkFlagsArray[@]}"}
-            ${checkTarget}
         )
+        : "${checkFlags:=VERBOSE=y}"
```

Comment:
I don’t think this is affects anything, but we did not assign default value for `checkFlags` globally before this change.

I’m not sure if this can be handled elegantly or whether if it has to be addressed at all. Perhaps
```bash
if [[ -v checkFlags ]]; then
    concatTo flagsArray checkFlags
else
    appendToVar flagsArray VERBOSE=y
fi
```
Same for other places where `${var:-default}` expansion was replaced with `${var:=default}` assignment.

Or set the default from `mkDerivation`.

>[!NOTE]
>Open for discussion, not a suggestion for change, not blocking.

Reply:
I prefer the explicit helper.

Reply:
I’m fine with either (as well as setting the default globally) since you’ve confirmed that nothing is affected by this change (https://github.com/NixOS/nixpkgs/pull/335666#discussion_r1722227995). Note that `local` variable is visible from child function calls, i.e. _post*_ hooks, but I don’t think that’s an issue.